test/pg-cdc/mzcompose.yml

```diff
@@ -32,13 +45,34 @@ services:
     mzbuild: materialized
     command: --experimental --disable-telemetry
     ports:
-      - 6875:6875
+      - 6875
     environment:
     - MZ_DEV=1
     - MZ_LOG
 
   postgres:
     image: postgres:11.4
     ports:
-      - 5432:5432
-    command: postgres -c wal_level=logical -c max_wal_senders=20 -c max_replication_slots=20
+      - 5432
+    volumes:
+      - secrets:/secrets
+      - ./pg_hba.conf:/pg_hba.conf
+    command: >
+      postgres
+        -c wal_level=logical
+        -c max_wal_senders=20
+        -c max_replication_slots=20
+        -c ssl=on
+        -c hba_file=/pg_hba.conf
+        -c ssl_cert_file=/secrets/postgres.crt
+        -c ssl_key_file=/secrets/postgres.key
+    depends_on: [postgres-prepare-certs]
+
+  # Postgres runs as non-root. Is there a better way to fix these perms?
+  postgres-prepare-certs:
```

Comment:
I'm not sure if it would make this setup simpler, but the postgres docker image supports adding initialization scripts in it that run automatically. `rust-postgres` uses this mechanism to set up certificates, `pg_hba.conf`, and users.

This is the script https://github.com/sfackler/rust-postgres/blob/master/docker/sql_setup.sh which is then imported here https://github.com/sfackler/rust-postgres/blob/master/docker/Dockerfile#L3

Reply:
Nice! I hadn't seen this before. I think it will make things quite a bit simpler in combination with mzbuild (assuming I can depend on `test-certs` directly). Let me try that real quick and update if possible.

Reply:
You can also check out ```test/kafka-ssl/mzcompose.yml``` where the testdrive container entrypoint is used to put certs into position.

Reply:
As discussed offline, the approach from `test/kafka-ssl/mzcompose.yml` does not carry over, because we run testdrive after postgres has started.

It took a bit longer than expected, but I was able to simplify the `mzcompose.yml` by introducing a `postgres` mzimage.